packages/@aws-cdk/core/lib/token.ts

```diff
@@ -7,6 +7,33 @@ import { TokenMap } from './private/token-map';
 import { IResolvable, ITokenResolver } from './resolvable';
 import { TokenizedStringFragments } from './string-fragments';
 
+/**
+ * An enum-like class that represents the result of comparing two Tokens.
+ * The return type of {@link Token.compareStrings}.
+ */
+export class TokenComparison {
```

Comment:
Why isn't this just a simple enum?

Reply:
Because it allows us to add helper methods to it later, like `isEqualOrUnresolved()` that were here before, which I still think is the correct way to model this.

If we make this an enum, we will lose that flexibility.

Reply:
Feels like "backwards-future-compatibility". Not sure I understand the use case of `isEqualOfUnresolved()`, can you elaborate or show an example?

Reply:
I mean in code like this:

```ts
        const thisAndPolicyAccountComparison = Token.compareStrings(this.resourceEnv.account, policy.resourceEnv.account);
        const equalOrAnyUnresolved = thisAndPolicyAccountComparison === TokenComparison.SAME ||
          thisAndPolicyAccountComparison === TokenComparison.BOTH_UNRESOLVED ||
          thisAndPolicyAccountComparison === TokenComparison.ONE_UNRESOLVED;
        if (equalOrAnyUnresolved) {
          // ...
```

I think a much more natural way to write that is:

```ts
        if (Token.compareStrings(this.resourceEnv.account, policy.resourceEnv.account)
            .isEqualOrAnyUnresolved()) {
          // ...
```

I don't want us to block ourselves from achieving the latter by using an enum.

Reply:
Feels a bit non idiomatic (fluent and such).

I would have done this:

```ts
Token. isEqualOrAnyUnresolved(this.resourceEnv.account, policy.resourceEnv.account)
```

But if you feel strongly, I am fine...